src/interactive/azext_interactive/azclishell/recommendation.py

```diff
@@ -237,59 +243,81 @@ def get_recommend_from_api(command_list, type, top_num=5, error_info=None):  # p
+    if processed_exception and processed_exception != '':
+        feedback_data["error_info"] = processed_exception
 
+    # get all recommend sources and types
     has_personalized_rec = False
     if recommends:
+        recommends_list = []
         source_list = set()
-        rec_type_list = set()
+        recommend_type_list = set()
         for item in recommends:
+            try:
+                recommends_list.append({"command": str(item['command'])})
+            except KeyError:
+                recommends_list.append({"scenario": str(item['scenario'])})
             source_list.add(str(item['source']))
-            rec_type_list.add(str(item['type']))
+            recommend_type_list.add(str(item['type']))
             if 'is_personalized' in item:
                 has_personalized_rec = True
-        feedback_data.append(' '.join(source_list))
-        feedback_data.append(' '.join(rec_type_list))
-    else:
-        feedback_data.extend([' ', ' '])
-
-    if rec:
-        feedback_data.append(str(rec['source']))
-        feedback_data.append(str(rec['type']))
-        if rec['type'] == RecommendType.Scenario:
-            feedback_data.extend([rec['scenario'], ' '])
+        feedback_data["recommendations_source_list"] = sorted(source_list)
+        feedback_data["recommendations_type_list"] = sorted(recommend_type_list)
+        feedback_data["recommendations_list"] = recommends_list
+
+    if accepted_recommend:
+        feedback_data["accepted_recommend_source"] = accepted_recommend['source']
+        feedback_data["accepted_recommend_type"] = accepted_recommend['type']
+        if accepted_recommend['type'] == RecommendType.Scenario:
+            feedback_data['accepted_recommend'] = accepted_recommend['scenario']
         else:
-            feedback_data.append(rec['command'])
-            if "arguments" in rec and rec["arguments"]:
-                feedback_data.append(' '.join(rec["arguments"]))
-            else:
-                feedback_data.append(' ')
+            feedback_data['accepted_recommend'] = accepted_recommend['command']
 
         if not has_personalized_rec:
-            feedback_data.extend([' '])
-        elif 'is_personalized' in rec:
-            feedback_data.extend(['1'])
+            feedback_data["is_personalized"] = None
+        elif 'is_personalized' in accepted_recommend:
+            feedback_data["is_personalized"] = 1
         else:
-            feedback_data.extend(['0'])
-    else:
-        feedback_data.extend([' ', ' ', ' ', ' ', ' '])
+            feedback_data["is_personalized"] = 0
+
+    # replace null to None:
+    for key, value in feedback_data.items():
+        if value is None:
+            feedback_data[key] = "None"
 
     telemetry.start(mode='interactive')
     telemetry.set_command_details('next')
-    telemetry.set_feedback("#".join(feedback_data))
+    telemetry.set_recommendation_properties(api_version=api_version, recommendation_properties=feedback_data)
```

Comment:
The `recommendation_properties` we are storing now is a dict, right?
If so, we should no longer worry about misalignment, so we no longer need to initialize the `feedback_data`